instrumentation/src/javatest/org/oppia/android/instrumentation/ExplorationPlayerTest.kt

```diff
@@ -0,0 +1,4631 @@
+package org.oppia.android.instrumentation
+
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiCollection
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiScrollable
+import androidx.test.uiautomator.UiSelector
+import androidx.test.uiautomator.Until.hasObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+
+/** Tests for Explorations. */
+class ExplorationPlayerTest {
```

Comment:
Where did these tests come from? Did you reference other test suites?

I actually think there are too many tests here. Per the testing triangle end-to-end tests are more expensive and slower to run, and thus there should be fewer of them. The advantage of these tests is they can very complex, whole-user journeys that unit tests are unable to verify. We should focus the tests we add to this suite on verifying whole user journeys rather than very specific details like whether a button is enabled.

For example, playing through an entire exploration would be a single test since it encapsulates a full user journey.

Note also that the standard 'arrange-act-assert' flow doesn't always fit for these kinds of tests. They often imply correctness simply by passing since that means the full flow worked correctly.

Reply:
removed